kubernetes-addons/keda/locals.tf

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: MIT-0
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this
+ * software and associated documentation files (the "Software"), to deal in the Software
+ * without restriction, including without limitation the rights to use, copy, modify,
+ * merge, publish, distribute, sublicense, and/or sell copies of the Software, and to
+ * permit persons to whom the Software is furnished to do so.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+locals {
+  keda_service_account_name = "keda-operator"
+  keda_namespace            = "keda-ns"
+
+  irsa_set_values = [{
```

Comment:
No need to use set. Just use `default_keda_helm_values`. You are setting the same values in different places.

Reply:
Ok. I thought about it a lot here and we need `set` here to ensure our `serviceAccount.create` and `serviceAccount.name` when `var.keda_create_irsa = true` 

`default_keda_helm_values` will be overwritten if customer sends his own values.yaml and we will loose control.

I am using set but i am also validating this before applying in `main.tf` line61. With this approach we can control better